tests/stdlib/tstats.nim

```diff
@@ -30,6 +30,38 @@ rs1.push(@[1.0, 2.2, 1.4, 4.9])
 doAssert(rs1.sum == 9.5)
 doAssert(rs1.mean() == 2.375)
 
+block:
+  var 
+    ASv = [7,4,6,3,9,1]        #  sorted   1 3 4 6 7 9
+    A   : array[len(ASv),int]
+
+  func myCmp(x,y:int):int =
+    if x==y : 0 elif x<y : -1 else: 1
+
+  A= ASv;  doAssert quickSelect(A, 2, 5, 2, myCmp) == 1  # smallest element
+  A= ASv;  doAssert quickSelect(A, 2, 5, 5, myCmp) == 9  # largest element
+  #
+  A= ASv;  doAssert median(A,0,5,myCmp) == 5.0
+  A= ASv;  doAssert median(A,0,5,mdlow,myCmp) == 4
+  A= ASv;  doAssert median(A,0,5,mdhigh,myCmp) == 6
+  #
+  A= ASv;  doAssert median(A,2,5) == 4.5
+  A= ASv;  doAssert median(A,2,5,mdlow) == 3
+  A= ASv;  doAssert median(A,2,5,mdhigh) == 6
+  #
+  A= ASv;  doAssert median(A,0,4,myCmp) == 6.0
+  A= ASv;  doAssert median(A,0,4,mdlow,myCmp) == 6
+  A= ASv;  doAssert median(A,0,4,mdhigh,myCmp) == 6
+
```

Comment:
lots of nep1 violations; please read nep1 and recently written tests for best practices

Reply:
Thanks for your careful proof-reading of this PR and your valuable suggestions.
I hope I have implemented these correctly.